packages/move/sources/vesting.move

```diff
@@ -1,285 +1,290 @@
+        let delete_ref = object::generate_delete_ref(&constructor_ref);
+        move_to(&object_signer, ObjectController { extend_ref, delete_ref });
 
         fungible_asset::transfer(
             sender,
-            primary_fungible_store::primary_store(sender_addr, vesting_pool_mut.fa_metadata_object),
-            vesting_pool_mut.vesting_store,
+            primary_fungible_store::primary_store(sender_addr, fa_metadata_object),
+            vesting_store,
             amount
         );
-    }
 
-    // ================================= View Functions ================================= //
-
-    #[view]
-    /// Get vesting data
-    public fun get_vesting_data(): (Object<Metadata>, Object<FungibleStore>, u64) acquires VestingData {
-        let vesting_data = borrow_global<VestingData>(@deployment_addr);
-        (vesting_data.fa_metadata_object, vesting_data.vesting_store, vesting_data.total_vesting_amount)
-    }
-
-    #[view]
-    /// Whether vesting stream exists
-    public fun exists_vesting_stream(beneficiary: address): bool acquires VestingData {
-        let vesting_data = borrow_global<VestingData>(@deployment_addr);
-        simple_map::contains_key(&vesting_data.streams, &beneficiary)
-    }
-
-    #[view]
-    /// Get vesting stream data
-    public fun get_vesting_stream(beneficiary: address): (u64, u64, u64, u64, u64) acquires VestingData {
-        let vesting_data = borrow_global<VestingData>(@deployment_addr);
-        let vesting_stream = simple_map::borrow(&vesting_data.streams, &beneficiary);
-        (
-            vesting_stream.amount,
-            vesting_stream.claimed_amount,
-            vesting_stream.start_time,
-            vesting_stream.cliff,
-            vesting_stream.duration
-        )
-    }
-
-    #[view]
-    public fun get_claimable_amount(user: address): u64 acquires VestingData {
-        // Check if the user has a vesting stream
-        // Check if the user has any tokens to claim
-
-        let vesting_data = borrow_global<VestingData>(@deployment_addr);
-        assert!(
-            simple_map::contains_key(&vesting_data.streams, &user),
-            ERR_VESTING_STREAM_DOES_NOT_EXIST
+        event::emit(
+            CreateVestingStreamEvent {
+                object_address: signer::address_of(&object_signer),
+                vesting_stream: object::address_to_object<VestingStream>(signer::address_of(&object_signer))
+            }
         );
-        let vesting_stream = simple_map::borrow(&vesting_data.streams, &user);
-        let claimable_amount =
-            calculate_vested_amount(
-                vesting_stream.amount,
-                vesting_stream.start_time,
-                vesting_stream.cliff,
-                vesting_stream.duration
-            );
-        claimable_amount - vesting_stream.claimed_amount
-
     }
 
+    // ================================= View Functions ================================= //
```

Comment:
it would be good to implement other view functions as well, for example a function that returns info about a stream(e.g. duration, claimed_amount, total_amount, etc)